## In this notobook, I have used LinearRegression to predict cost and time based on different parameters.

In [1]:
#importing necessary libraries
import pandas as pd
import numpy as np
import sklearn
import warnings
import re
warnings.filterwarnings('ignore')

In [2]:
#necessary modeule

%load_ext autoreload
%autoreload 2

In [3]:
#reading dataset
data=pd.read_csv('/kaggle/input/nepal-travel-data/Data.csv')

In [4]:
#dropping unwanted columns
data=data.drop(columns='Unnamed: 0')

In [5]:
#changing time column datatype
data['Time'] = data['Time'].apply(lambda x: re.sub(r'\b[Dd]ays?\b', '', x)).astype(int)

In [6]:
#changing cost column datatype
data['Cost']=data['Cost'].str.replace('$','').str.replace('USD','').str.replace(',','').str.replace('\n','').astype(int)

In [7]:
#changing altitude column datatype
data['Max Altitude']=data['Max Altitude'].str.replace('m','').str.replace(',','').astype(int)

In [8]:
#rechecking datatypes
data.dtypes

Trek                         object
Cost                          int64
Time                          int64
Trip Grade                   object
Max Altitude                  int64
Accomodation                 object
Best Travel Time             object
Contact or Book your Trip    object
dtype: object

#### Below, I have predicted total Cost required in trek in user input Time.

In [9]:
#taking variable x as Time and y as Cost
x=data['Time'].values.reshape(-1,1)
y=data['Cost'].values.reshape(-1,1)

In [10]:
#importing necessary libraries required in LinearRegression
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression

In [11]:
#splitting data into testing and training phase
X_train,X_test,y_train,y_test=train_test_split(x,y,random_state=42,test_size=0.8)

In [12]:
#defining and fitting linear regression
reg=LinearRegression()
reg.fit(X_train,y_train)

LinearRegression()

In [13]:
#predicting result 
y_pred=reg.predict(X_test)

In [14]:
y_pred=reg.predict([[22]])
y_pred

array([[1596.12487654]])

In [15]:
y_pred=reg.predict([[10]])
y_pred

array([[1149.93048482]])

**Observation: Here,we found that if we take 22 days as travel time , it will cost us total of $1596 as travel cost.**

**Similarly, if we take 10 days as travel time, it will cost us total of $1149 as travel cost.**

In [16]:
#taking varaible z as max altitude and v as time
z=data['Max Altitude'].values.reshape(-1,1)
v=data['Time'].values.reshape(-1,1)

In [17]:
#splitting the data into training and testing phase
X_train,X_test,y_train,y_test=train_test_split(z,v,test_size=0.8,random_state=42)

In [18]:
#predicting results
u_pred=reg.predict(X_test)

In [19]:
u_pred=reg.predict([[15]])
u_pred

array([[1335.8448147]])

In [20]:
u_pred=reg.predict([[10]])
u_pred

array([[1149.93048482]])

**Observation: If we take 15 days as travel time , then we can climb altitude of 1135m and if we take 10 days as travel time,then we can only climb altitude of 1149m**

# Finally we have finished applying LinearRegression